pkg/controllers/management/auth/globalroles/fleetworkspace_binding_handler.go

```diff
@@ -0,0 +1,195 @@
+	globalRole, err := h.grCache.Get(globalRoleBinding.GlobalRoleName)
+	if err != nil {
+		return fmt.Errorf("unable to get globalRole: %w", err)
+	}
+	if globalRole.InheritedFleetWorkspacePermissions.WorkspaceVerbs == nil &&
+		globalRole.InheritedFleetWorkspacePermissions.ResourceRules == nil {
+		return nil
+	}
+
+	if err = h.reconcileResourceRulesBindings(globalRoleBinding, globalRole); err != nil {
+		return fmt.Errorf("error reconciling fleet permissions rules: %w", err)
+	}
+
+	if err = h.reconcileWorkspaceVerbsBindings(globalRoleBinding, globalRole); err != nil {
+		return fmt.Errorf("error reconciling fleet workspace verbs: %w", err)
+	}
+
+	return nil
+}
+
+func (h *fleetWorkspaceBindingHandler) reconcileResourceRulesBindings(grb *v3.GlobalRoleBinding, gr *v3.GlobalRole) error {
+	if gr.InheritedFleetWorkspacePermissions.ResourceRules == nil {
+		return nil
+	}
+	fleetWorkspaces, err := h.fwCache.List(labels.Everything())
+	if err != nil {
+		return fmt.Errorf("unable to list fleetWorkspaces when reconciling globalRoleBinding %s: %w", grb.Name, err)
+	}
+
+	var returnError error
+	for _, fleetWorkspace := range fleetWorkspaces {
+		if fleetWorkspace.Name == localFleetWorkspace {
+			continue
+		}
+		rb, err := h.rbCache.Get(fleetWorkspace.Name, wrangler.SafeConcatName(grb.Name))
+		if err != nil && !apierrors.IsNotFound(err) {
+			returnError = multierror.Append(returnError, err)
+			continue
+		}
+		desiredRB := backingRoleBinding(grb, gr, fleetWorkspace.Name)
+		if apierrors.IsNotFound(err) {
+			_, err = h.rbClient.Create(desiredRB)
+			if err != nil && !apierrors.IsNotFound(err) {
+				returnError = multierror.Append(returnError, err)
+			}
+		} else if !reflect.DeepEqual(rb.Subjects, desiredRB.Subjects) ||
+			!reflect.DeepEqual(rb.RoleRef, desiredRB.RoleRef) {
+			// undo modifications if rb has changed.
+			err := h.rbClient.Delete(rb.Namespace, rb.Name, &metav1.DeleteOptions{})
+			if err != nil {
+				returnError = multierror.Append(returnError, err)
+			}
+			_, err = h.rbClient.Create(desiredRB)
+			if err != nil {
+				returnError = multierror.Append(returnError, err)
+			}
+		}
+	}
+
+	return returnError
+}
+
+func (h *fleetWorkspaceBindingHandler) reconcileWorkspaceVerbsBindings(grb *v3.GlobalRoleBinding, gr *apimgmtv3.GlobalRole) error {
+	if gr.InheritedFleetWorkspacePermissions.WorkspaceVerbs == nil {
+		return nil
+	}
+	crbName := wrangler.SafeConcatName(grb.Name, fleetWorkspaceVerbsName)
+	crb, err := h.crbCache.Get(crbName)
+	if err != nil && !apierrors.IsNotFound(err) {
+		return err
+	}
+
+	desiredCRB := backingClusterRoleBinding(grb, gr, crbName)
+	if apierrors.IsNotFound(err) {
+		_, err = h.crbClient.Create(desiredCRB)
+
+		return err
+	}
+	if !reflect.DeepEqual(crb.Subjects, desiredCRB.Subjects) ||
+		!reflect.DeepEqual(crb.RoleRef, desiredCRB.RoleRef) {
+		// undo modifications if crb has changed.
+		err := h.crbClient.Delete(crb.Name, &metav1.DeleteOptions{})
+		if err != nil {
```

Comment:
Similar comment here about checking for a not found error.

Reply:
done